src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/subquery/rewriter/SubqueryAliasRewriter.java

```diff
@@ -0,0 +1,148 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.parser.subquery.rewriter;
+
+import com.alibaba.druid.sql.ast.SQLExpr;
+import com.alibaba.druid.sql.ast.expr.SQLAllColumnExpr;
+import com.alibaba.druid.sql.ast.expr.SQLBinaryOpExpr;
+import com.alibaba.druid.sql.ast.expr.SQLBinaryOperator;
+import com.alibaba.druid.sql.ast.expr.SQLIdentifierExpr;
+import com.alibaba.druid.sql.ast.expr.SQLInSubQueryExpr;
+import com.alibaba.druid.sql.ast.statement.SQLExprTableSource;
+import com.alibaba.druid.sql.ast.statement.SQLSelectItem;
+import com.alibaba.druid.sql.dialect.mysql.ast.statement.MySqlSelectQueryBlock;
+import com.alibaba.druid.sql.dialect.mysql.visitor.MySqlASTVisitorAdapter;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * Add the alias for identifier the subquery query.
+ * Use the table alias if it already has one, Auto generate if it doesn't has one.
+ */
+public class SubqueryAliasRewriter extends MySqlASTVisitorAdapter {
+
+    private Deque<Table> tableScope =  new ArrayDeque<>();
+
+    private int aliasSuffix = 0;
+
+    private final static String DOT = ".";
+
+    @Override
+    public boolean visit(MySqlSelectQueryBlock query) {
+        final SQLExprTableSource from = (SQLExprTableSource) query.getFrom();
+        String tableName = from.getExpr().toString().replaceAll(" ", "");
+
+        if (from.getAlias() != null) {
+            tableScope.push(new Table(tableName, from.getAlias()));
+        } else {
+            from.setAlias(createAlias(tableName));
+            tableScope.push(new Table(tableName, from.getAlias()));
+        }
+
+        for (SQLSelectItem item : query.getSelectList()) {
+            final SQLExpr itemExpr = item.getExpr();
+            if (itemExpr instanceof SQLIdentifierExpr) {
```

Comment:
I think it would be cleaner to implement `visit(SQLIdentifierExpr)` because I notice you're rewriting it in multiple places in this class. Does the following sample code make sense?

```
    @Override
    public boolean visit(SQLIdentifierExpr expr) {
        if (isInSelect(expr) || isInWhere(expr) || isInSubquery(expr) {
            rewrite(tableScope.peek(), expr);
        }
        return true;
    }
```

Reply:
Good suggestion! Done.